src/blueprint/operators/call.rs

```diff
@@ -0,0 +1,202 @@
+  args
+    .clone()
+    .find_map(|(k, value)| if k == key { Some(value) } else { None })
+}
+
+pub fn update_call(
+  operation_type: &GraphQLOperationType,
+) -> TryFold<'_, (&Config, &Field, &config::Type, &str), FieldDefinition, String> {
+  TryFold::<(&Config, &Field, &config::Type, &str), FieldDefinition, String>::new(
+    move |(config, field, _, _), b_field| {
+      let Some(call) = &field.call else {
+        return Valid::succeed(b_field);
+      };
+
+      compile_call(field, config, call, operation_type)
+        .and_then(|resolver| Valid::succeed(b_field.resolver(Some(resolver))))
+    },
+  )
+}
+
+pub fn compile_call(
+  field: &Field,
+  config: &Config,
+  call: &config::Call,
+  operation_type: &GraphQLOperationType,
+) -> Valid<Expression, String> {
+  Valid::from_option(call.query.clone(), "call must have query".to_string())
+    .and_then(|field_name| {
+      Valid::from_option(config.find_type("Query"), "Query type not found on config".to_string())
+        .zip(Valid::succeed(field_name))
+    })
+    .and_then(|(query_type, field_name)| {
+      Valid::from_option(
+        query_type.fields.get(&field_name),
+        format!("{} field not found", field_name),
+      )
+      .zip(Valid::succeed(field_name))
+      .and_then(|(field, field_name)| {
+        if field.has_resolver() {
+          Valid::succeed((field, field_name, call.args.iter()))
+        } else {
+          Valid::fail(format!("{} field has no resolver", field_name))
+        }
+      })
+    })
+    .and_then(|(_field, field_name, args)| {
+      let empties: Vec<(&String, &config::Arg)> = _field
+        .args
+        .iter()
+        .filter(|(k, _)| !args.clone().any(|(k1, _)| k1.eq(*k)))
+        .collect();
+
+      if empties.len().gt(&0) {
+        return Valid::fail(format!(
+          "no argument {} found",
+          empties
+            .iter()
+            .map(|(k, _)| format!("'{}'", k))
+            .collect::<Vec<String>>()
+            .join(", ")
+        ))
+        .trace(field_name.as_str());
+      }
+
+      if let Some(http) = _field.http.clone() {
+        compile_http(config, field, &http).and_then(|expr| match expr.clone() {
+          Expression::IO(IO::Http { req_template, group_by, dl_id }) => Valid::succeed(
+            req_template
+              .clone()
+              .root_url(replace_url(&req_template.root_url, &args)),
+          )
+          .map(|req_template| {
+            req_template
+              .clone()
+              .query(req_template.clone().query.iter().map(replace_mustache(&args)).collect())
+          })
+          .map(|req_template| {
+            req_template
+              .clone()
+              .headers(req_template.headers.iter().map(replace_mustache(&args)).collect())
+          })
+          .map(|req_template| Expression::IO(IO::Http { req_template, group_by, dl_id })),
+          _ => Valid::succeed(expr),
```

Comment:
this code and the others `_ => Valid::succeed(expr)` on this file are unreachable. can I delete it @tusharmath @meskill?